tools/cmake/FindBpfProg.cmake

```diff
@@ -0,0 +1,188 @@
+# SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#[=======================================================================[.rst:
+FindBpfProg
+--------
+
+Find BpfProg
+
+This module finds if all the dependencies for eBPF Compile-Once-Run-Everywhere
+programs are available and where all the components are located.
+
+The caller may set the following variables to disable automatic
+search/processing for the associated component:
+
+  ``BPFPROG_BPFTOOL_EXE``
+    Path to ``bpftool`` binary
+
+  ``BPFPROG_CLANG_EXE``
+    Path to ``clang`` binary
+
+  ``LIBBPF_INCLUDE_DIRS``
+    Path to ``libbpf`` development headers
+
+  ``LIBBPF_LIBRARIES``
+    Path to `libbpf` library
+
+  ``BPFPROG_VMLINUX_H``
+    Path to ``vmlinux.h`` generated by ``bpftool``. If unset, this module will
+    attempt to automatically generate a copy.
+
+This module sets the following result variables:
+
+::
+
+  BpfProg_FOUND             = TRUE if all components are found
+
+
+This module also provides the ``bpf_prog()`` macro. This macro generates a
+cmake interface library for the BPF program's generated skeleton as well
+as the associated dependencies.
+
+.. code-block:: cmake
+
+  bpf_prog(<name> <source>)
+
+Given an abstract ``<name>`` for a BPF program and the associated ``<source>``
+file, generates an interface library target, ``<name>_skel``, that may be
+linked against by other cmake targets.
+
+Example Usage:
+
+::
+
+  find_package(BpfProg REQUIRED)
+  bpf_prog(myprog myprog.bpf.c)
+  add_executable(myapp myapp.c)
+  target_link_libraries(myapp myprog_skel)
+
+#]=======================================================================]
+
+if(NOT BPFPROG_BPFTOOL_EXE)
+  find_program(BPFPROG_BPFTOOL_EXE NAMES bpftool DOC "Path to bpftool executable")
+endif()
+
+if(NOT BPFPROG_CLANG_EXE)
+  find_program(BPFPROG_CLANG_EXE NAMES clang DOC "Path to clang executable")
+
+  execute_process(COMMAND ${BPFPROG_CLANG_EXE} --version
+    OUTPUT_VARIABLE CLANG_version_output
+    ERROR_VARIABLE CLANG_version_error
+    RESULT_VARIABLE CLANG_version_result
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  # Check that clang is new enough
+  if(${CLANG_version_result} EQUAL 0)
+    if("${CLANG_version_output}" MATCHES "^clang version ([^\n]+)\n")
```

Comment:
The version output could be something like:
```
Ubuntu clang version 11.0.0-2~ubuntu20.04.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
which has a `Ubuntu ` prefix, I think we could remove the `^`.

Reply:
I modify this locally, and it works now. Thanks.

Reply:
Ah yeah. Annoying how distros change version output.